src/swap/v2/router/swap-router.ts

```diff
@@ -0,0 +1,198 @@
+import algosdk, {ALGORAND_MIN_TX_FEE, getApplicationAddress, Transaction} from "algosdk";
+import AlgodClient from "algosdk/dist/types/src/client/v2/algod/algod";
+
+import {CONTRACT_VERSION} from "../../../contract/constants";
+import {ALGO_ASSET_ID} from "../../../util/asset/assetConstants";
+import {getAssetId, isAlgo} from "../../../util/asset/assetUtils";
+import {SupportedNetwork} from "../../../util/commonTypes";
+import {getValidatorAppID} from "../../../validator";
+import {SwapType} from "../../constants";
+import {
+  FetchSwapRouteQuotesPayload,
+  FetchSwapRouteQuotesResponse,
+  GenerateSwapRouterTxnsParams
+} from "../../types";
+import {
+  V2_SWAP_APP_CALL_ARG_ENCODED,
+  V2_SWAP_APP_CALL_SWAP_TYPE_ARGS_ENCODED,
+  V2_SWAP_ROUTER_APP_ARGS_ENCODED
+} from "../constants";
+import {TINYMAN_ANALYTICS_API_BASE_URLS} from "./constants";
+import {getSwapRouterAppID} from "./util";
+
+export async function generateSwapRouterAssetOptInTransaction({
+  client,
+  routerAppID,
+  assetIDs,
+  accountAddress
+}: {
+  client: AlgodClient;
+  routerAppID: number;
+  assetIDs: number[];
+  accountAddress: string;
+}): Promise<Transaction> {
+  const suggestedParams = await client.getTransactionParams().do();
+  // We need to create a NoOp transaction for each asset we want to opt-in to
```

Comment:
I'm a bit confused about this comment, it says "we need to create txn for each asset".
I understand we need N txns for N assets, but I guess this is not correct, because we just create one txn here 🤔

Reply:
Yes the comment is wrong we need to create only one transaction for this purpose